java/data/worldgen/mod.mcdoc

```diff
@@ -43,30 +43,41 @@ type IntProvider<T> = (
 	T |
 	struct {
 		type: #[id="int_provider_type"] string,
-		#[until="1.20.5"]
-		value: minecraft:int_provider[[type]]<T>,
-		#[since="1.20.5"]
 		...minecraft:int_provider[[type]]<T>,
 	} |
 )
 
 type ConstantIntProvider<T> = struct {
 	value: T,
 }
-dispatch minecraft:int_provider[constant] to ConstantIntProvider
+
+dispatch minecraft:int_provider[constant] to (
+	ConstantIntProvider |
+	#[until="1.20.5"] struct {
+		value: ConstantIntProvider
+	} |
+)
+
+type IntProviderSettings<S, T> = (
```

Comment:
I'm not sure if this actually works: the dispatching only provides S, while T is supposed to be provided after indexing.

Reply:
I haven't looked at this in detail, but I don't think so. When you use type parameters is overwrites the old type mapping

Reply:
It might just be cleaner to have this `value` wrapping in each necessary int provider

Reply:
I'll test it myself tomorrow. If it doesn't work we'll have to find a different solution. Its a bit of an annoying situation that the `weighted_list` int provider doesn't get its config wrapped in a `value` field pre 1.20.5.

Reply:
> It might just be cleaner to have this `value` wrapping in each necessary int provider

I've done this now. Makes it much easier to read as well. I've also done the same for the particles.